tools/jdbc/src/main/java/org/duckdb/DuckDBTimestamp.java

```diff
@@ -0,0 +1,68 @@
+package org.duckdb;
+
+import java.sql.Timestamp;
+import java.time.ZoneOffset;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+public class DuckDBTimestamp {
+    static 
+    {
+        // LocalDateTime reference of epoch
+        RefLocalDateTime = 
+            LocalDateTime.ofEpochSecond(0, 0, ZoneOffset.UTC);
+    }
+
+    public DuckDBTimestamp(long timeMicros) {
+        this.timeMicros = timeMicros;
+    }
+
+    public DuckDBTimestamp(Timestamp sqlTimestamp) {
+        this.timeMicros = DuckDBTimestamp.RefLocalDateTime.until(
```

Comment:
DuckDB (and RDBMSes in general) store GMT instants, so is local time what is wanted here?

Reply:
Java's sql.Timestamp and LocalDateTime (and util.Date) do not have a timezone, but internally they do have it in the difference to epoch. For storage, this has to be normalized to GMT and translated to the same value in the client timezone while reading.

JDBC has only get and setTimestamp, but get and setObject allow to ask for a LocalDateTime. For this reason the transformations are already implemented, though the get/setObject calls for LocalDateTime are not supported (yet).

Reply:
Ah, so older Java uses naïve timestamps?

Reply:
If you mean "just store the diff to epoch as int" with naïve timestamps, then I believe the answer is no.